pkg/controller/datavolume-controller.go

```diff
@@ -2565,3 +2578,81 @@ func GetRequiredSpace(filesystemOverhead float64, requestedSpace int64) int64 {
 func newLongTermCloneTokenGenerator(key *rsa.PrivateKey) token.Generator {
 	return token.NewGenerator(common.ExtendedCloneTokenIssuer, key, 10*365*24*time.Hour)
 }
+
+func (r *DatavolumeReconciler) garbageCollect(dataVolume *cdiv1.DataVolume, pvc *corev1.PersistentVolumeClaim) (res *reconcile.Result, err error) {
+	if dataVolume.Status.Phase != cdiv1.Succeeded {
+		return
+	}
+	cdiConfig := &cdiv1.CDIConfig{}
+	if err = r.client.Get(context.TODO(), types.NamespacedName{Name: common.ConfigName}, cdiConfig); err != nil {
+		return
+	}
+	dvTTL := cdiConfig.Spec.DataVolumeTTLSeconds
+	if dvTTL == nil || *dvTTL < 0 {
+		return
+	}
+
+	// Current DV still has TTL, so reconcile will return with the needed RequeueAfter
+	if delta := getDeltaTTL(dataVolume, *dvTTL); delta > 0 {
+		return &reconcile.Result{RequeueAfter: delta}, nil
+	}
+
+	if err = r.detachPvcDeleteDv(pvc, dataVolume); err != nil {
+		return
+	}
+
+	// FIXME: not sure this part is necessary
+	// Lookup for other DVs to garbage collect
```

Comment:
Yeah same here, I am not sure if other DVs won't be queued up for their own loop and just get GC'd there

Reply:
Yep, I think this part should be removed.